stdlib/public/core/Range.swift.gyb

```diff
@@ -493,24 +340,15 @@ ${get_init_warning(Self, OtherSelf)}
   public init(_ other: ${OtherSelf}<Bound>) {
 %   if 'Closed' not in Self and 'Closed' in OtherSelf:
     let upperBound = other.upperBound.advanced(by: 1)
-%   elif 'Closed' in Self and 'Closed' not in OtherSelf:
+%   else:
     _precondition(!other.isEmpty, "Can't form an empty closed range")
     let upperBound = other.upperBound.advanced(by: -1)
-%   else:
-    let upperBound = other.upperBound
```

Comment:
Don't we still need this logic for the same-type initialisers (i.e `Range`'s `init(_: Range)` & `ClosedRange`'s `init(_: ClosedRange)`)? Otherwise we'd be decrementing the upper bound in those cases. Or is the plan to get rid of the same-type initialisers?

Reply:
Yeah, this looks like a meta programming fail on my part. I'm going to de-gyb the whole thing in the next push.

There's no reason to do a `Range.init` from a `Range`, in fact. It should just be a deprecation warning (for people who were using it to convert the now-typealiased `CountableRange` into a `Range`).